File: bridget_mcmahon/week10/tuesday/integers/sum_integers.rb
# def sum_integers(filename)
#   sum = 0
#
#   file = File.new filename
#   file.each_line do |line|
#     sum += line.to_i
#   end
#
#   sum
# end

def sum_integers(filename)
  File.readlines(filename).map(&:to_i).inject(:+)
end

puts sum_integers('integers.txt')

# require 'pry'
# binding.pry

File: bridget_mcmahon/week10/tuesday/integers/sumIntegers.js
const fs = require('fs'); // File Systems code from Node std library

// const sumIntegers = function (filename) {
//   let sum = 0;
//   const file = fs.readFileSync(filename, 'utf-8');
//   file.split('\n').forEach(function(line) {
//     const n = parseInt(line);
//     sum += n;
//   });
//   return sum;
// };
//
// console.log(sumIntegers('./integers.txt'));

const sumIntegers2 = function(filename) {
  let sum = 0;
  const file = fs.readFileSync(filename, 'utf-8');

  return file.split('\n').reduce(function(acc, n) {
    return Number(acc) + Number(n);
  });
}

console.log(sumIntegers2('./integers.txt'));
